app-backend/tile/src/main/scala/TileIO.scala

```diff
@@ -0,0 +1,262 @@
+package com.azavea.rf.tile.project
+
+import java.util.UUID
+
+import com.azavea.rf.common.cache._
+import com.azavea.rf.common.cache.kryo.KryoMemcachedClient
+import com.azavea.rf.common.{KamonTraceRF, Config => CommonConfig}
+import com.azavea.rf.database.Database
+import com.azavea.rf.database.tables.ScenesToProjects
+import com.azavea.rf.datamodel.MosaicDefinition
+
+import akka.dispatch.MessageDispatcher
+import cats.data._
+import cats.implicits._
+import com.typesafe.scalalogging.LazyLogging
+import geotrellis.proj4._
+import geotrellis.raster._
+import geotrellis.raster.histogram._
+import geotrellis.raster.io._
+import geotrellis.slick.Projected
+import geotrellis.spark._
+import geotrellis.spark.io._
+import geotrellis.spark.io.postgres.PostgresAttributeStore
+import geotrellis.spark.io.s3._
+import geotrellis.vector._
+import spray.json.DefaultJsonProtocol._
+
+import scala.concurrent._
+import scala.util._
+
+/** Trait with various IO operations for tiles and related metadata */
+trait TileIO extends LazyLogging with KamonTraceRF {
+
+  implicit val database: Database
+  implicit val blockingDispatcher: MessageDispatcher
+  val memcachedClient: KryoMemcachedClient
+  val rfCache: CacheClient
```

Comment:
This is kind of a general comment: Once there are `val` fields the trait cake pattern used here becomes invalid. While the functions are separated out into their own files, there is still no clear sense of who owns the fields and why. 

For purpose of clarifying the component dependency it would be more clear to turn traits into classes and place all the vals into constructor arguments. The client code can be kept the same by doing imports from the class instances to get all the functions in scope.